packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,199 @@
+const componentName: string = "route-search";
+
+/**
+ * Component to search a particular route with autocompletion that will redirect to the route chosen by the user
+ */
+@Component({
+	selector: "route-search",
+	templateUrl: "./route-search.component.html",
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: componentName
+	}
+})
+export class StarkRouteSearchComponent implements OnInit {
+	
+	/**
+	 * the controller of the form
+	 */
+	public pathCtrl: FormControl;
+
+	/**
+	 * the list of routes filtered by autocomplete
+	 */
+	public filteredOptions: Observable<any[]>;
+
+	/**
+	 * the list of routes filtered by autocomplete
+	 */
+	public optionsToDisplay: RouteEntry[] = [];
+	
+	/**
+	 * The menu which options will be displayed
+	 */
+	@Input()
+	public optionsList?: StarkMenuConfig;
+	
+	/**
+	 * the direction in which we want to input field to be displayed 
+	 */
+	@Input()
+	public direction: StarkRouteSearchDirection = "left";
+	
+	public inputValue: string ="";
+	
+	/**
+	 * if the input field must be hidden or not
+	 */
+	public hide: boolean;
+
+	public constructor(@Inject(STARK_ROUTING_SERVICE) public starkRoutingService: StarkRoutingService,
+					   @Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+					   @Inject(UIRouter) public routingService: UIRouter) {
+		this.pathCtrl = new FormControl();
+	}
+	
+	/**
+	 * Filters the options according to what has been entered by the user in the input field
+	 */
+	public filterOptions(value: string): RouteEntry[] {
+		const filterValue: string = value.toLowerCase();
+		return this.optionsToDisplay.filter((state: RouteEntry) => state.label.toLowerCase().indexOf(filterValue) === 0);
+	}
+
+	
+	public ngOnInit(): void {
+
+		this.inputValue="";
+		this.filteredOptions = this.pathCtrl.valueChanges.pipe(
+			startWith(""),
+			map((option: string) => (option ? this.filterOptions(option) : this.optionsToDisplay.slice()))
+		);
+		
+		this.hide = true;
+		
+		this.logger.debug(componentName + ": component initialized");
+
+		if (this.optionsToDisplay !== undefined){
+			if (this.optionsList === undefined){
+				this.optionsToDisplay =	this.initRoutePathFromRouter();
+			} else{
+				this.optionsToDisplay = this.initRoutePathFromConfig();
+			}
+		}
+		this.sortPathList();
+		this.inputValue="";
+	}
+
+	/**
+	 * redirect the user to the chosen path
+	 */
+	public redirect(value: string): void {
+		this.inputValue="";
+		this.starkRoutingService.navigateTo(value);
+	}
+
+	/**
+	 * Show/hide the input field
+	 */
+	public show(): void {
+		this.hide = !this.hide;
+	}
+	/**
+	 * @ignore
+	 */
+	public trackPath(_index: number, path: RouteEntry): string {
+		if (path.targetState !== undefined){
+			return path.targetState;
+		} else 
+		{
+			return "";
+		}
+	}
+
+	/**
+	 * Retrieve the list of routes from a StarkMenuConfig object which the user have passed to the component
+	 */
+	//tslint:disable:cognitive-complexity
+	public initRoutePathFromConfig(): RouteEntry[]{
```

Comment:
Shouldn't this function be recursive? Because the StarkMenuConfig is recursive as well...